### PR TITLE
Checkbox validation markup (Take 2)

### PR DESCRIPTION
**Pull request by mdo**
Cherry-picking the commit from #17141 since that PR was fubared.

/cc @zacechola

### CODE REVIEW

File: docs/components/forms.md [hunk 1]
```diff
@@ -584,6 +580,11 @@ Ensure that an alternative indication of state is also provided. For instance, y
       Checkbox with error
```
Comment (cvrebert) [inline docs/components/forms.md]: with *danger*

File: docs/components/forms.md [hunk 2]
```diff
@@ -584,6 +580,11 @@ Ensure that an alternative indication of state is also provided. For instance, y
...
     </label>
   </div>
+<div class="checkbox has-danger">
+  <label>
+    <input type="checkbox" id="checkboxError" value="option1">
+    Checkbox with error
```
Comment (cvrebert) [inline docs/components/forms.md]: with *danger*

**Comment by mdo** [comment]
Gah, fixing those comments. Damn autoupdates came in too late.

**Comment by mdo** [comment]
Follow ups in f712e92 and 925f0ca. Also noticed some jank that I fixed in 9e23bbc to unfuck some broken docs content.

**Pull request merged**